src/js/polyfills/worker.js

```diff
@@ -1,15 +1,42 @@
 const core = globalThis[Symbol.for('tjs.internal.core')];
 const _Worker = core.Worker;
 
+import { kBlobGetParts } from './blob';
 import { defineEventAttribute } from './event-target';
+import { kGetObjectURL } from './url';
 
 const kWorker = Symbol('kWorker');
 
+function blobTextSync(blob) {
+    const decoder = new TextDecoder();
+    const partsStr = [];
+
+    for (const part of blob[kBlobGetParts]) {
+        if (part instanceof Blob) {
+            partsStr.push(blobTextSync(part));
+        } else {
+            partsStr.push(decoder.decode(part, { stream: true }));
+        }
+    }
+
+    partsStr.push(decoder.decode());
+
+    return partsStr.join('');
+}
+
 class Worker extends EventTarget {
     constructor(path) {
         super();
 
-        const worker = new _Worker(path);
+        let source = undefined;
+
+        const blob = URL[kGetObjectURL](path);
+
+        if (blob) {
```

Comment:
I am wondering if for the readability, and preparing your future support for data URL, it wouldn't be better to have a function to check if path is an objecturl, if so, then get the objecturl. In the end, this adds an unnecessary step, but it may make the code more expressive?

Reply:
I agree. Basically parse the path as a URL, and check the scheme.

Reply:
Hum, a problem is that currently, when passing a path to the worker, since it does not have an url pattern, putting that path inside a new URL object raise an error. So, as an alternative, I used `URLPattern`.

Reply:
That is fine! You'll want to do something like:

```
let url;
try {
    url = new URL(specifier);
} catch (_) {
    // It's not a URL, it's a path.
}
...
```